OneSignalSDK/onesignal/core/src/main/java/com/onesignal/user/internal/state/UserState.kt

```diff
@@ -0,0 +1,27 @@
+package com.onesignal.user.internal.state
+
+import org.json.JSONObject
+
+/**
+ * A user state.
+ */
+class UserState (
```

Comment:
The majority of the external API is defined through interfaces, even for simple POJO types like this ([example](https://github.com/OneSignal/OneSignal-Android-SDK/blob/user-model/main/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/notifications/IActionButton.kt)), to solidify an understanding of what is intended as an external interface.

I think it's okay to define a class as part of the external interface, but keep in mind anything that goes in here could be interpreted to be part of the API.